File: Java/DPR.java
/*Saya Muhammad Alfi faiz NIM 2207045 mengerjakan
soal Latihan 1 dalam mata kuliah Desain Pemograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/


// Membuat kelas DPR
public class DPR {
  
  // Atribut privat informasi anggota.
  private String nama;   // Nama anggota
  private String partai; // partai politik anggota
  private String bidang; // Bidang anggota
  private int id;        // id anggota

  // Konstruktor default
  public DPR() {
      this.nama = "";
      this.partai = "";
      this.bidang = "";
      this.id = 0;
  }

  // Konstruktor berparameter
  public DPR(int id, String nama, String bidang, String partai) {
      this.nama = nama;
      this.partai = partai;
      this.bidang = bidang;
      this.id = id;
  }

  // Metode mengambil nama anggota.
  public String getNama() {
      return this.nama;
  }

  // Metode memperbarui nama anggota.
  public void setNama(String nama) {
      this.nama = nama;
  }

  // Metode mengambil data partai politik.
  public String getPartai() {
      return this.partai;
  }

  // Metode memperbarui data partai politik.
  public void setPartai(String partai) {
      this.partai = partai;
  }

  // Metode mengambil bidang.
  public String getBidang() {
      return this.bidang;
  }

  // Metode memperbarui bidang.
  public void setBidang(String bidang) {
      this.bidang = bidang;
  }

  // Metode mengambil id.
  public int getId() {
      return this.id;
  }

  // Metode memperbarui id.
  public void setId(int id) {
      this.id = id;
  }
}

File: PHP/DPR.php
<?php
/*Saya Muhammad Alfi faiz NIM 2207045 mengerjakan
soal Latihan 1 dalam mata kuliah Desain Pemograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

class DPR
{
    private $nama;
    private $partai;
    private $bidang;
    private $id;
    private $foto; // tambahan properti foto

    public function __construct($id, $nama, $partai, $bidang, $foto)
    {
        $this->nama = $nama;
        $this->partai = $partai;
        $this->bidang = $bidang;
        $this->id = $id;
        $this->foto = $foto;
    }

    public function get_nama()
    {
        return $this->nama;
    }

    public function set_nama($nama)
    {
        $this->nama = $nama;
    }

    public function get_partai()
    {
        return $this->partai;
    }

    public function set_partai($partai)
    {
        $this->partai = $partai;
    }

    public function get_bidang()
    {
        return $this->bidang;
    }

    public function set_bidang($bidang)
    {
        $this->bidang = $bidang;
    }

    public function get_id()
    {
        return $this->id;
    }

    public function set_id($id)
    {
        $this->id = $id;
    }

    public function get_foto()
    {
        return $this->foto;
    }

    public function set_foto($foto)
    {
        $this->foto = $foto;
    }

    public function __destruct()
    {
    }
}
?>
